esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -919,15 +964,29 @@ void EsdlServiceImpl::handleServiceRequest(IEspContext &context,
         }
     }
 
-    context.addTraceSummaryTimeStamp(LogNormal, "srt-resLogging");
-    handleResultLogging(context, scriptContext, tgtctx.get(), req, soapmsg.str(), origResp.str(), out.str(), logdata.str());
-    context.addTraceSummaryTimeStamp(LogNormal, "end-resLogging");
     ESPLOG(LogMax,"Customer Response: %s", out.str());
 }
 
-bool EsdlServiceImpl::handleResultLogging(IEspContext &espcontext, IEsdlScriptContext *scriptContext, IPropertyTree * reqcontext, IPropertyTree * request,const char *rawreq, const char * rawresp, const char * finalresp, const char * logdata)
+bool EsdlServiceImpl::handleResultLogging(IEspContext &espcontext, IEsdlScriptContext *scriptContext, IEsdlDefService &srvdef, IEsdlDefMethod &mthdef, IPropertyTree * reqcontext, IPropertyTree * request,const char *rawreq, const char * rawresp, const char * finalresp, const char * logdata)
 {
+    StringBuffer temp;
+    if (scriptContext)
+    {
+        IEsdlTransformSet *servicePLTs = m_transforms->queryMethodEntryPoint("", ESDLScriptEntryPoint_PreLogging);
+        IEsdlTransformSet *methodPLTs = m_transforms->queryMethodEntryPoint(mthdef.queryName(), ESDLScriptEntryPoint_PreLogging);
+
+        scriptContext->appendContent(ESDLScriptCtxSection_LogData, "LogDataset", logdata);
+
+        if (servicePLTs || methodPLTs)
+        {
+            processServiceAndMethodTransforms(scriptContext, {servicePLTs, methodPLTs}, ESDLScriptCtxSection_LogData, nullptr);
+            scriptContext->toXML(temp, ESDLScriptCtxSection_PreESDLResponse);
+            logdata = temp.str();
```

Comment:
@afishbeck I don't understand what this should accomplish. PreESDLResponse does not appear to be populated after processing the transforms. It results in logdata=nullptr, and then the LogDatasets element is not available to the LogAgents because it is set on line 1000 below.

Did you mean to serialize `ESDLScriptCtxSection_LogData` here? If so then the LogDatasets will be passed to the LogAgents correctly if we also make the spelling change on line 978.

If we intend to pass the contents of PreESDLResponse to the agents as the LogDatasets, then we'll need to concatenate the LogData section also, or we'll need to see why the PreESDLResponse is not populated in this case.

Reply:
Yes, typo should be ESDLScriptCtxSection_LogData section being serialized.